bin/check_samplesheet.py

```diff
@@ -81,118 +171,32 @@ def check_samplesheet(file_in, file_out):
-            given = ",".join(header)
-            wanted = ",".join(MIN_HEADER)
-            print(f"ERROR: Please check samplesheet header -> {given} != {wanted}")
+    required_columns = {"sample", "fastq_1", "fastq_2"}
+    # See https://docs.python.org/3.9/library/csv.html#id3 to read up on `newline=""`.
+    with file_in.open(newline="") as in_handle:
+        # reader = csv.DictReader(in_handle, dialect=sniff_format(in_handle))
+        reader = csv.DictReader(in_handle)
+        # Validate the existence of the expected header columns.
+        if not required_columns.issubset(reader.fieldnames):
+            req_cols = ", ".join(required_columns)
+            logger.critical(f"The sample sheet **must** contain these column headers: {req_cols}.")
             sys.exit(1)
-
-        ## Check sample entries
-        for line in fin:
-            lspl = [x.strip().strip('"') for x in line.strip().split(",")]
-
-            # Check valid number of columns per row
-            if len(lspl) < len(header):
-                print_error(
-                    "Invalid number of columns (minimum = {})!".format(len(header)),
-                    "Line",
-                    line,
-                )
-            num_cols = len([x for x in lspl if x])
-            if num_cols < MIN_COLS:
-                print_error(
-                    "Invalid number of populated columns (minimum = {})!".format(MIN_COLS),
-                    "Line",
-                    line,
-                )
-
-            ## Check sample name entries
-            sample, fastq_1, fastq_2 = lspl[: len(MIN_HEADER)]
-            sample = sample.replace(" ", "_")
-            if not sample:
-                print_error("Sample entry has not been specified!", "Line", line)
-
-            ## Check expected cells is an integer if present
-            expected_cells = ""
-            if "expected_cells" in header:
-                expected_cells = lspl[colmap["expected_cells"]]
-                if not is_integer(expected_cells):
-                    print_error("Expected cells must be an integer", "Line", line)
-
-            ## If present, replace spaces with _ in sequencing center name
-            seq_center = ""
-            if "seq_center" in header:
-                seq_center = lspl[colmap["seq_center"]]
-                seq_center = seq_center.replace(" ", "_")
-
-            ## Check FastQ file extension
-            for fastq in [fastq_1, fastq_2]:
-                if fastq:
-                    if fastq.find(" ") != -1:
-                        print_error("FastQ file contains spaces!", "Line", line)
-                    if not fastq.endswith(".fastq.gz") and not fastq.endswith(".fq.gz"):
-                        print_error(
-                            "FastQ file does not have extension '.fastq.gz' or '.fq.gz'!",
-                            "Line",
-                            line,
-                        )
```

Comment:
These checks seem to be missing from the new version, or am I missing something?

Reply:
fastq is still done, but the other two have to be re-included, I mistakenly took it out.

Reply:
I have reverted it back to how it was. To the same format it was before, now, I just included some new checkings for the new columns following the same pattern as before.